csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -758,31 +764,42 @@ public sealed class Settings
             /// <summary>
             /// Default settings, as used by <see cref="JsonFormatter.Default"/>
             /// </summary>
-            public static Settings Default { get; }
+            public static Settings Default { get; } = new Settings();
 
-            // Workaround for the Mono compiler complaining about XML comments not being on
-            // valid language elements.
-            static Settings()
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object.
+            /// </summary>
+            public Settings() : this(false, TypeRegistry.Empty)
             {
-                Default = new Settings(false);
             }
 
             /// <summary>
             /// Whether fields whose values are the default for the field type (e.g. 0 for integers)
             /// should be formatted (true) or omitted (false).
             /// </summary>
-            public bool FormatDefaultValues { get; }
+            public bool FormatDefaultValues { get; set; }
 
+            private TypeRegistry typeRegistry_;
             /// <summary>
             /// The type registry used to format <see cref="Any"/> messages.
             /// </summary>
-            public TypeRegistry TypeRegistry { get; }
+            public TypeRegistry TypeRegistry {
+                get { return typeRegistry_ ; }
+                set {
+                    typeRegistry_ = ProtoPreconditions.CheckNotNull(value, nameof(value));
+                }
+            }
+
+            /// <summary>
+            /// Whether to format enums as ints. Defaults to false.
+            /// </summary>
+            public bool FormatEnumAsInt { get; set;}
 
             // TODO: Work out how we're going to scale this to multiple settings. "WithXyz" methods?
 
             /// <summary>
             /// Creates a new <see cref="Settings"/> object with the specified formatting of default values
-            /// and an empty type registry.
+            /// and an empty type registry. This method is deprecated.
```

Comment:
Please add the `[Obsolete]` attribute here and below, if we still deprecated these. (And call them constructors rather than methods.)